examples/js-sub.c

```diff
@@ -130,21 +171,48 @@ int main(int argc, char **argv)
 
     if (s == NATS_OK)
     {
-        if (pull)
+        if (pull && async)
+        {
+            jsOpts.PullSubscribeAsync.MaxMessages = (int) total;
```

Comment:
This is different to what all other clients do.

As per https://github.com/nats-io/nats-architecture-and-design/blob/main/adr/ADR-37.md

We treat `no_wait == true` as a footgun that can have heavy impact on the server.

Reply:
@Jarema please note that this PR/API should not be seen as the C version of JS simplification. It's main purpose is to fit in with the existing C APIs, to enable async usage pattern of the existing (`js_PullSubscribe`/`js_Fetch`/`js_FetchRequest`). The most immediate issue this PR solves is enabling the use of the thread pool for Pull subscriptions message delivery. Previously, the user had to start a thread of their own to accomplish this. So, this implementation attempts to stay idiomatic with the above (and other Subscribe) APIs.

Also, I did not attempt to de-couple the "Subscribe" vs [multiple] "Consume" operations because of how complex the subscription state logic already is in the C client. So, there is only one lifetime, that of the Fetch, and the sub is auto-unsubscribe if it's reached, or if a serious error happens(*see below).

As to "NoWait" - again, the idea was to enable the existing functionality (including `js_FetchRequest` which allows direct control of `NoWait`) in the async use case, using a thread pool. This was the rationale for keeping it visible.

Reply:
Discussed offline.